sha256-avx2/sha512x4.h

```diff
@@ -0,0 +1,40 @@
+#ifndef SHA512AVX_H
+#define SHA512AVX_H
+#include <stdint.h>
+#include "immintrin.h"
+
+typedef struct SHA512state4x {
+    __m256i s[8];
+    unsigned char msgblocks[4*128];
+    int datalen;
+    unsigned long long msglen;
+} sha512ctx4x;
```

Comment:
We do not need the incremental API: a single function that combines init_from_bytes; update and final will be faster. Not a blocker for the initial PR, but we should create an issue to remember.